core/dbt/parser/manifest.py

```diff
@@ -926,6 +929,33 @@ def process_sources(self, current_project: str):
                 continue
             _process_sources_for_exposure(self.manifest, current_project, exposure)
 
+    def process_nodes(self):
+        # make sure the nodes are in the manifest.nodes or the disabled dict,
+        # correctly now that the schema files are also aprsed
+        disable_nodes = []
+        for node in self.manifest.nodes.values():
+            if not node.config.enabled:
+                disable_nodes.append(node)
+        for node in disable_nodes:
+            self.manifest.nodes.pop(node.unique_id)
+            self.manifest.add_disabled_nofile(node)
+
+        enable_nodes = {}
+        for disabled in self.manifest.disabled.values():
+            for node in disabled:
+                if node.config.enabled:
+                    for dis_index, dis_node in enumerate(disabled):
+                        # Remove node from disabled and unique_id from disabled dict if necessary
+                        enable_nodes[dis_index] = dis_node
```

Comment:
I commented on this last time, but it looks like github lost it somehow...  Making the index the key in the 'enable_nodes' dictionary means that if you have multiple enabled nodes at the same index, the first one will be overwritten.

I'm not sure why you are saving these in separate structures. It seems like for both looping through the nodes and looping through the disabled you could just move them when encountered.

Reply:
@gshank I'm saving them as a separate structure because python doesn't like when you modify the dict you are looping through.

`disabled` on line 947 is the list of nodes for the unique id.  Probably worth renaming if it's confusing.  So the index is just the index of the list item.  There can't be multiple nodes at a single index since each index represents a single node.